java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToVarHint.java

```diff
@@ -160,4 +167,36 @@ private static boolean isDiagnosticCodeTobeSkipped(CompilationInfo info) {
         List<Diagnostic> diagnosticsList = info.getDiagnostics();
         return diagnosticsList.stream().anyMatch((d) -> (SKIPPED_ERROR_CODES.contains(d.getCode())));
     }
+    
+    private static boolean isValidVarType(HintContext ctx) {
+        TreePath treePath = ctx.getPath();
+        TreePath initTreePath = ctx.getVariables().get("$init");  //NOI18N
+
+        TypeMirror variableTypeMirror = ctx.getInfo().getTrees().getElement(treePath).asType();
+
+        if (initTreePath.getLeaf().getKind() == Tree.Kind.NEW_CLASS) {
```

Comment:
For consideration: maybe a switch instead of an if cascade?

Reply:
Done